articles/the-evolution-of-resource-sharing-technologies-over-the-internet/index.md

```diff
@@ -0,0 +1,48 @@
+
+ * **SOAP Web Services**
+
+    SOAP stands for Simple Object Access Protocol. It is a messaging protocol for interchanging data in a decentralized and distributed environment. SOAP works with many application-layer protocols, namely, HTTP, SMTP, TCP, or UDP. SOAP relies exclusively on the XML format to provide messaging services. The protocol is built with Security, authorization, and error-handling in mind.
+
+    SOAP is more suitable for enterprise-level web services that require high security and transactions that are highly complex. SOAP is commonly used in financial services, payment gateways, telecommunication services, etc. It most common API being PayPal's API, which allows payment through PayPal services or credit card.
+
+* **REST Web Services**
+
+    REST stands for Representational State Transfer. It is relatively new and has quickly become very popular. Developers usually find SOAP burdensome to use. For example, working with SOAP in JavaScript means converting even the simplest of tasks into the required XML structure.
+
+    On the other hand, REST allows API providers to deliver data in multiple formats such as plain text, HTML, XML, YAML, and JSON. This is one of the most loved features of REST. Due to the human-readable format and being lightweight, REST has gained a lot of traction. Despite its name, we can use JSON with any programming language - not just JavaScript.
+
+    REST, however, can only use the HTTP protocol for data transmission. REST also does not prescribe how to implement its principles at a lower level. Instead, it provides guidelines in the form of architectural principles or constraints but does not enforce them. Web services built following the REST architectural principles are called RESTful web services. To create a REST API, you need to follow five architectural principles:
+
+    1. **Uniform Interface** - Requests from different clients should look the same. That is, the same resources should not have more than one URI.
+    2. **Client-Server Separation** - The client and the server should not act as one but should be independent, which means that their only form of interaction is through requests and responses.
+    3. **Statelessness** - There should not be any server-side sessions. A single request should contain all the information the server needs to know.
+    4. **Cacheable resources** - Server responses should contain information about whether the data they send is cacheable or not.
+    5. **Layered System** - There might be several layers of servers between the client and the server that returns the response. This should not affect either the request or the response.
+---
```

Comment:
Conclude the article. Create a section called Conclusion, and summarize everything in the article.

Reply:
Got it, I'll work on it and update. Thank you